libmscore/figuredbass.cpp

```diff
@@ -1770,6 +1773,256 @@ FiguredBass* Score::addFiguredBass()
+                        harmony = native_pitch + int(AccidentalVal::FLAT);
+                        break;
+                  case FiguredBassItem::Modifier::NATURAL:
+                        harmony = native_pitch;
+                        break;
+                  case FiguredBassItem::Modifier::SHARP:
+                        harmony = native_pitch + int(AccidentalVal::SHARP);
+                        break;
+                  case FiguredBassItem::Modifier::DOUBLESHARP:
+                        harmony = native_pitch + int(AccidentalVal::SHARP);
+                        harmony = native_pitch + int(AccidentalVal(modifier));
+                        break;
+                  case FiguredBassItem::Modifier::CROSS:
+                  case FiguredBassItem::Modifier::BACKSLASH:
+                        harmony = native_pitch + int(acciv) + 1; // adding 1 semi tone, flat->natural, natural->sharp, etc.
+                        break;
+                  case FiguredBassItem::Modifier::SLASH:
+                  case FiguredBassItem::Modifier::NUMOF:
+                  default:
+                        qDebug("measure #%d Modifier = %hhd not yet implemented", segment->measure()->no(), modifier);
+                        return;
+            }
+            harmony += octave;
+            for ( int h : harmonys )
+                  while ( h >= harmony )
+                        harmony += 12;
+            harmonys.push_back(harmony);
+      };
+      for (Element* e : segment->annotations()) {
+            if (e->type() == Element::Type::FIGURED_BASS ) {
+                  FiguredBass *fb = static_cast<FiguredBass*>(e);
+                  std::vector<int> harmony;
+                  std::vector<FiguredBassItem*> items = fb->getFiguredBaseItems();
+                  sort( items.begin(), items.end(), [&](FiguredBassItem *i1, FiguredBassItem *i2) {
+                        return get_digit(i1->digit()) < get_digit(i2->digit());
+                  });
+                  FiguredBassItem::Modifier modifier = (none == items[0]->prefix()) ? items[0]->suffix(): items[0]->prefix();
+                  switch ( items.size() ) {
+                        case 0:
+                              render_harmony(none, 3, none);
+                              render_harmony(none, 5, none);
+                              break;
+                        case 1: {
+                              // a single figured bass missing a number, just with a symbol, implies the digit 3.
+                              int digit = (FBIDigitNone==items[0]->digit()) ? 3 : items[0]->digit();
+                              switch (digit) {
+                                    case 2:
+                                          // render_relative({2,4,6});
+                                          render_harmony(modifier,2,none);
+                                          render_harmony(none, 4, none);
+                                          render_harmony(none, 6, none);
```

Comment:
Again, not sure if you are a wikipedia contributor, but if the article is wrong, you might consider updating it?
http://en.wikipedia.org/wiki/Figured_bass <http://en.wikipedia.org/wiki/Figured_bass>

> On 24 May 2015, at 05:20, Maurizio M. Gavioli <notifications@github.com> wrote:
> 
> In libmscore/figuredbass.cpp <https://github.com/musescore/MuseScore/pull/2015#discussion_r30949810>:
> 
> > +                  });
> > +                  FiguredBassItem::Modifier modifier = (none == items[0]->prefix()) ? items[0]->suffix(): items[0]->prefix();
> > +                  switch ( items.size() ) {
> > +                        case 0:
> > +                              render_harmony(none, 3, none);
> > +                              render_harmony(none, 5, none);
> > +                              break;
> > +                        case 1: {
> > +                              // a single figured bass missing a number, just with a symbol, implies the digit 3.
> > +                              int digit = (FBIDigitNone==items[0]->digit()) ? 3 : items[0]->digit();
> > +                              switch (digit) {
> > +                                    case 2:
> > +                                          // render_relative({2,4,6});
> > +                                          render_harmony(modifier,2,none);
> > +                                          render_harmony(none, 4, none);
> > +                                          render_harmony(none, 6, none);
> Hmmm... 2 alone usually means 2 3 5
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/musescore/MuseScore/pull/2015/files#r30949810>.
>